### ISSUE TITLE
Broken installation of OSE 3.7.x as of 2018-03-21

### ISSUE THREAD

**Issue by jperville** [IshentRas/cookbook-openshift3#219]
Hello everyone,

2 days ago, packages for OSE v3.7.1 were pushed to http://mirror.centos.org/centos/7/paas/x86_64/openshift-origin37/, however these new v3.7.1 packages are completely broken and install a version which identifies as v3.10 alpha (see https://github.com/openshift/openshift-ansible/issues/7567#issuecomment-374455481).

After re-initializing my deployment and locking the `ose_version` explicitly to `3.7.0-1.0.7ed6862`, the cookbook starts installing the correct version but when installing the origin-node package it somehow tries to install the broken v3.7.1 version instead of the v3.7.0 I requested, as if one of the `package` resource in this cookbook is not passing the `ose_version` in the `version` attribute.

The broken version of origin-node will trigger installation of the broken version of the origin-master packages, resulting in the converge failing with the following output:

```
==> master: [2018-03-21T08:11:51+00:00] ERROR: yum_package[origin-sdn-ovs] (cookbook-openshift3::node line 105) had an error: Chef::Exceptions::Exec: yum -d0 -e0 -y install origin-sdn-ovs-3.7.0-1.0.7ed6862 returned 1:
==> master: STDOUT:  You could try using --skip-broken to work around the problem
==> master:  You could try running: rpm -Va --nofiles --nodigest
==> master: 
==> master: STDERR: Error: Package: origin-sdn-ovs-3.7.0-1.0.7ed6862.x86_64 (centos-openshift-origin37)
==> master:            Requires: origin-node = 3.7.0-1.0.7ed6862
==> master:            Installed: origin-node-3.7.1-1.el7.git.0.0a2d6a1.x86_64 (@centos-openshift-origin37)
==> master:                origin-node = 3.7.1-1.el7.git.0.0a2d6a1
==> master:            Available: origin-node-3.7.0-1.0.7ed6862.x86_64 (centos-openshift-origin37)
==> master:                origin-node = 3.7.0-1.0.7ed6862
==> master: [2018-03-21T08:11:51+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

**Comment by IshentRas** [comment]
@jperville I know and I have already got a fix for that shit :-) SHould be pushed ASAP. I have already got it here

**Comment by IshentRas** [comment]
I had to fix it here: https://github.com/IshentRas/cookbook-openshift3/blob/master/attributes/default.rb#L34
and here: https://github.com/IshentRas/cookbook-openshift3/blob/master/recipes/validate.rb#L21-L23

**Comment by jperville** [comment]
This is ugly indeed.

**Comment by IshentRas** [comment]
Revert my changes as they have released a fix version `3.7.1-2`.
Please close if it is now ok on your side. (It should be)

**Comment by IshentRas** [comment]
Closing as solved
